src/consumption.cpp

```diff
@@ -101,69 +101,150 @@ int player::stomach_capacity() const
-    float kcal = 0;
+    assert( comest.get_comestible() );
 
-    // if item has components, will derive calories from that instead.
-    if( !comest.components.empty() && !comest.has_flag( "NUTRIENT_OVERRIDE" ) ) {
-        int byproduct_multiplier;
-        for( const item &component : comest.components ) {
-            component.has_flag( "BYPRODUCT" ) ? byproduct_multiplier = -1 : byproduct_multiplier = 1;
-            kcal += this->kcal_for( component ) * component.charges * byproduct_multiplier;
-        }
-        kcal /= comest.recipe_charges;
-    } else {
-        kcal = comest.get_comestible()->get_calories();
+    // As float to avoid rounding too many times
+    float kcal = comest.get_comestible()->default_nutrition.kcal;
 
-        // Many raw foods give less calories, as your body has expends more energy digesting them.
-        // We don't want RAW to stack for components and results, so we're doing it in this else block.
-        if( comest.has_flag( "RAW" ) && !comest.has_flag( "COOKED" ) ) {
-            kcal *= 0.75f;
-        }
+    // Many raw foods give less calories, as your body has expends more energy digesting them.
+    if( comest.has_flag( "RAW" ) && !comest.has_flag( "COOKED" ) ) {
+        kcal *= 0.75f;
     }
 
-    if( has_trait( trait_GIZZARD ) ) {
+    if( p.has_trait( trait_GIZZARD ) ) {
         kcal *= 0.6f;
     }
 
-    if( has_trait( trait_CARNIVORE ) && comest.has_flag( flag_CARNIVORE_OK ) &&
+    if( p.has_trait( trait_CARNIVORE ) && comest.has_flag( flag_CARNIVORE_OK ) &&
         comest.has_any_flag( carnivore_blacklist ) ) {
         // TODO: Comment pizza scrapping
         kcal *= 0.5f;
     }
 
     const float relative_rot = comest.get_relative_rot();
     // Saprophages get full nutrition from rotting food
-    if( relative_rot > 1.0f && !has_trait( trait_SAPROPHAGE ) ) {
+    if( relative_rot > 1.0f && !p.has_trait( trait_SAPROPHAGE ) ) {
         // everyone else only gets a portion of the nutrition
         // Scaling linearly from 100% at just-rotten to 0 at halfway-rotten-away
         const float rottedness = clamp( 2 * relative_rot - 2.0f, 0.1f, 1.0f );
         kcal *= ( 1.0f - rottedness );
     }
 
     // Bionic digestion gives extra nutrition
-    if( has_bionic( bio_digestion ) ) {
+    if( p.has_bionic( bio_digestion ) ) {
         kcal *= 1.5f;
     }
 
     return static_cast<int>( kcal );
 }
 
+// list of traits the player has that modifies vitamin absorption
+static std::list<trait_id> mut_vitamin_absorb_modify( const player &p )
+{
+    std::list<trait_id> traits;
+    for( auto &m : p.get_mutations() ) {
+        const auto &mut = m.obj();
+        if( !mut.vitamin_absorb_multi.empty() ) {
+            traits.push_back( m );
+        }
+    }
+    return traits;
+}
+
+// is the material associated with this item?
+static bool material_exists( const material_id &material, const item &item )
+{
+    for( const material_id &mat : item.type->materials ) {
```

Comment:
The whole function can be reduced to just two lines:
```C++
    const auto &mats = item.type->materials;
    return std::find( mats.begin(), mats.end(), mat ) != mats.end();
```

Reply:
In fact, this function already exists: `item::made_of( const material_id & )` - use that one instead.

Reply:
No wonder it's hard to find with a name like that. `item::has_material` or something else with the word material in it would be way more likely to be found with code completion.